include/zephyr/llext/llext.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LLEXT_H
+#define ZEPHYR_LLEXT_H
+
+#include <zephyr/sys/slist.h>
+#include <zephyr/llext/elf.h>
+#include <zephyr/llext/symbol.h>
+#include <zephyr/llext/loader.h>
+#include <sys/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Linkable loadable extensions
+ * @defgroup llext Linkable loadable extensions
+ * @ingroup os_services
+ * @{
+ */
+
+/**
+ * @brief Enum of memory regions for lookup tables
+ */
+enum llext_mem {
+	LLEXT_MEM_TEXT,
+	LLEXT_MEM_DATA,
+	LLEXT_MEM_RODATA,
+	LLEXT_MEM_BSS,
+
+	LLEXT_MEM_COUNT,
+};
+
+/**
+ * @brief Linkable loadable extension
+ */
+struct llext {
+	sys_snode_t _llext_list;
```

Comment:
missing docstring

Reply:
This is meant to be a private member and I've wrapped it in ```@cond ignore``` but as other comments here noted, if there's a better way to signal this I'm happy to adjust